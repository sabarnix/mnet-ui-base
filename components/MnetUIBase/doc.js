"use strict";

exports.__esModule = true;
exports.themeDoc = exports.doc = void 0;

var _reactDesc = require("react-desc");

var _utils = require("../../utils");

var doc = function doc(MnetUIBase) {
  var DocumentedMnetUIBase = (0, _reactDesc.describe)(MnetUIBase).availableAt((0, _utils.getAvailableAtBadge)('MnetUIBase')).description('The top level MnetUIBase container.').usage("import { MnetUIBase } from 'mnet-ui-base';\n<MnetUIBase>...</MnetUIBase>").intrinsicElement('div');
  DocumentedMnetUIBase.propTypes = {
    background: _utils.backgroundDoc,
    dir: _reactDesc.PropTypes.oneOf(['rtl']).description('Layout direction for right to left contexts'),
    full: _reactDesc.PropTypes.bool.description('Whether to take the whole viewport.').defaultValue(false),
    plain: _reactDesc.PropTypes.bool.description("Whether or not MnetUIBase should apply a global font-family, font-size,\n        and line-height.").defaultValue(false),
    cssVars: _reactDesc.PropTypes.bool.description('Whether to expose the css variables.').defaultValue(false),
    theme: _reactDesc.PropTypes.object.description('Custom styles for MnetUIBase app component.'),
    themeMode: _reactDesc.PropTypes.oneOf(['dark', 'light']).description("Dark vs. light theme variation. Default is unspecified and left to\n      theme."),
    userAgent: _reactDesc.PropTypes.string.description("User agent used to detect the device width for setting the initial\n      breakpoint.")
  };
  return DocumentedMnetUIBase;
};

exports.doc = doc;
var themeDoc = {
  'mnet.extend': {
    description: 'Any additional style for MnetUIBase.',
    type: 'string | (props) => {}',
    defaultValue: undefined
  },
  'global.font.face': {
    description: 'Custom font face declaration',
    type: 'string | (props) => {}',
    defaultValue: undefined
  }
};
exports.themeDoc = themeDoc;