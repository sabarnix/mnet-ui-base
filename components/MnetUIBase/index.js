"use strict";

exports.__esModule = true;
exports.MnetUIBase = void 0;

var _MnetUIBase = require("./MnetUIBase");

exports.MnetUIBase = _MnetUIBase.MnetUIBase;