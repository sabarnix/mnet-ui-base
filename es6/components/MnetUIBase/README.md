## MnetUIBase
The top level MnetUIBase container.

[![](https://cdn-images-1.medium.com/fit/c/120/120/1*TD1P0HtIH9zF0UEH28zYtw.png)](https://storybook.grommet.io/?selectedKind=MnetUIBase&full=0&addons=0&stories=1&panelRight=0) [![](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/github/grommet/grommet-sandbox?initialpath=mnetuibase&module=%2Fsrc%2FMnetUIBase.js)
## Usage

```javascript
import { MnetUIBase } from 'mnet-ui-base';
<MnetUIBase>...</MnetUIBase>
```

## Properties

**background**

Either a color 
identifier to use for the background color. For example: 'neutral-1'. Or, a 
'url()' for an image. Dark is not needed if color is provided.

```
string
{
  color: 
    string
    {
      dark: string,
      light: string
    },
  dark: 
    boolean
    string,
  image: string,
  position: string,
  opacity: 
    string
    boolean
    number
    weak
    medium
    strong,
  repeat: 
    no-repeat
    repeat
    string,
  size: 
    cover
    contain
    string,
  light: string
}
```

**dir**

Layout direction for right to left contexts

```
rtl
```

**full**

Whether to take the whole viewport.

```
boolean
```

**plain**

Whether or not MnetUIBase should apply a global font-family, font-size,
        and line-height.

```
boolean
```

**cssVars**

Whether to expose the css variables.

```
boolean
```

**theme**

Custom styles for MnetUIBase app component.

```
object
```

**themeMode**

Dark vs. light theme variation. Default is unspecified and left to
      theme.

```
dark
light
```

**userAgent**

User agent used to detect the device width for setting the initial
      breakpoint.

```
string
```
  
## Intrinsic element

```
div
```
## Theme
  
**mnet.extend**

Any additional style for MnetUIBase. Expects `string | (props) => {}`.

Defaults to

```
undefined
```

**global.font.face**

Custom font face declaration Expects `string | (props) => {}`.

Defaults to

```
undefined
```
