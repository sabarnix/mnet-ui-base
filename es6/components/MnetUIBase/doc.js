import { describe, PropTypes } from 'react-desc';
import { backgroundDoc, getAvailableAtBadge } from '../../utils';
export var doc = function doc(MnetUIBase) {
  var DocumentedMnetUIBase = describe(MnetUIBase).availableAt(getAvailableAtBadge('MnetUIBase')).description('The top level MnetUIBase container.').usage("import { MnetUIBase } from 'mnet-ui-base';\n<MnetUIBase>...</MnetUIBase>").intrinsicElement('div');
  DocumentedMnetUIBase.propTypes = {
    background: backgroundDoc,
    dir: PropTypes.oneOf(['rtl']).description('Layout direction for right to left contexts'),
    full: PropTypes.bool.description('Whether to take the whole viewport.').defaultValue(false),
    plain: PropTypes.bool.description("Whether or not MnetUIBase should apply a global font-family, font-size,\n        and line-height.").defaultValue(false),
    cssVars: PropTypes.bool.description('Whether to expose the css variables.').defaultValue(false),
    theme: PropTypes.object.description('Custom styles for MnetUIBase app component.'),
    themeMode: PropTypes.oneOf(['dark', 'light']).description("Dark vs. light theme variation. Default is unspecified and left to\n      theme."),
    userAgent: PropTypes.string.description("User agent used to detect the device width for setting the initial\n      breakpoint.")
  };
  return DocumentedMnetUIBase;
};
export var themeDoc = {
  'mnet.extend': {
    description: 'Any additional style for MnetUIBase.',
    type: 'string | (props) => {}',
    defaultValue: undefined
  },
  'global.font.face': {
    description: 'Custom font face declaration',
    type: 'string | (props) => {}',
    defaultValue: undefined
  }
};