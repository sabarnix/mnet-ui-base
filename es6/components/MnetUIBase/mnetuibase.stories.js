import React from 'react';
import { storiesOf } from '@storybook/react';
import { mnet, MnetUIBase, Anchor, Box } from 'mnet-ui-base';
import { Add } from "grommet-icons/es6/icons/Add";
var customTheme = {
  global: {
    colors: {
      custom: '#cc6633'
    }
  }
};

var Themed = function Themed() {
  return /*#__PURE__*/React.createElement(MnetUIBase, {
    theme: customTheme
  }, /*#__PURE__*/React.createElement(Box, {
    pad: "medium"
  }, /*#__PURE__*/React.createElement(Anchor, {
    icon: /*#__PURE__*/React.createElement(Add, null),
    label: "Add",
    color: "custom"
  })));
};

var Plain = function Plain() {
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(MnetUIBase, {
    plain: true
  }, /*#__PURE__*/React.createElement(Box, {
    pad: "medium"
  }, /*#__PURE__*/React.createElement("p", null, "Plain MnetUIBase"))), /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(Box, {
    pad: "medium"
  }, /*#__PURE__*/React.createElement("p", null, "Not plain MnetUIBase"))));
};

var MnetUIBaseVars = function MnetUIBaseVars() {
  return /*#__PURE__*/React.createElement(MnetUIBase, {
    theme: mnet,
    cssVars: true
  }, /*#__PURE__*/React.createElement(Box, {
    pad: "medium",
    background: "var(--accent-2)",
    gap: "medium"
  }, /*#__PURE__*/React.createElement(Box, null, "Checkout MnetUIBase variables, you can find them in the StyledMnetUIBase DOM."), /*#__PURE__*/React.createElement(Box, {
    "with": true
  }, "For example, the background color in this Box is using var(--accent-2)")));
};

storiesOf('MnetUIBase', module).add('Plain', function () {
  return /*#__PURE__*/React.createElement(Plain, null);
}).add('Theme', function () {
  return /*#__PURE__*/React.createElement(Themed, null);
}).add('Vars', function () {
  return /*#__PURE__*/React.createElement(MnetUIBaseVars, null);
});