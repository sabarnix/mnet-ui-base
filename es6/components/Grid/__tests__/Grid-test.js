import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import { MnetUIBase } from '../../MnetUIBase';
import { Grid } from '..';
describe('Grid', function () {
  test('renders', function () {
    var component = renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(Grid, null)));
    var tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('a11yTitle renders', function () {
    var _render = render( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(Grid, {
      a11yTitle: "My Grid"
    }))),
        container = _render.container,
        getByLabelText = _render.getByLabelText;

    var gridWithLabel = getByLabelText('My Grid');
    expect(gridWithLabel).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
  test('rows renders', function () {
    var component = renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(Grid, {
      rows: ['small', 'large', 'medium']
    }), /*#__PURE__*/React.createElement(Grid, {
      rows: "small"
    })));
    var tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('columns renders', function () {
    var component = renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(Grid, {
      columns: ['1/2', '2/4']
    }), /*#__PURE__*/React.createElement(Grid, {
      columns: ['1/3', '2/3']
    }), /*#__PURE__*/React.createElement(Grid, {
      columns: ['1/4', '3/4']
    }), /*#__PURE__*/React.createElement(Grid, {
      columns: "small"
    }), /*#__PURE__*/React.createElement(Grid, {
      columns: {
        count: 'fit',
        size: 'small'
      }
    }), /*#__PURE__*/React.createElement(Grid, {
      columns: {
        count: 'fill',
        size: ['small', 'medium']
      }
    })));
    var tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('areas renders', function () {
    var component = renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(Grid, {
      rows: ['xxsmall', 'medium', 'xsmall'],
      columns: ['3/4', '1/4'],
      areas: [{
        name: 'header',
        start: [0, 0],
        end: [0, 1]
      }, {
        name: 'main',
        start: [1, 0],
        end: [1, 0]
      }, {
        name: 'sidebar',
        start: [1, 1],
        end: [1, 1]
      }, {
        name: 'footer',
        start: [2, 0],
        end: [2, 1]
      }]
    })));
    var tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('areas renders when given an array of string arrays', function () {
    var component = renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(Grid, {
      rows: ['xxsmall', 'medium', 'xsmall'],
      columns: ['3/4', '1/4'],
      areas: [['header', 'header'], ['sidebar', 'main'], ['footer', 'footer']]
    })));
    var tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('justify renders', function () {
    var component = renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(Grid, {
      justify: "start"
    }), /*#__PURE__*/React.createElement(Grid, {
      justify: "center"
    }), /*#__PURE__*/React.createElement(Grid, {
      justify: "end"
    }), /*#__PURE__*/React.createElement(Grid, {
      justify: "stretch"
    })));
    var tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('align renders', function () {
    var component = renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(Grid, {
      align: "start"
    }), /*#__PURE__*/React.createElement(Grid, {
      align: "center"
    }), /*#__PURE__*/React.createElement(Grid, {
      align: "end"
    }), /*#__PURE__*/React.createElement(Grid, {
      align: "stretch"
    })));
    var tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('justifyContent renders', function () {
    var component = renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(Grid, {
      justifyContent: "start"
    }), /*#__PURE__*/React.createElement(Grid, {
      justifyContent: "center"
    }), /*#__PURE__*/React.createElement(Grid, {
      justifyContent: "between"
    }), /*#__PURE__*/React.createElement(Grid, {
      justifyContent: "around"
    }), /*#__PURE__*/React.createElement(Grid, {
      justifyContent: "end"
    }), /*#__PURE__*/React.createElement(Grid, {
      justifyContent: "stretch"
    })));
    var tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('alignContent renders', function () {
    var component = renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(Grid, {
      alignContent: "start"
    }), /*#__PURE__*/React.createElement(Grid, {
      alignContent: "center"
    }), /*#__PURE__*/React.createElement(Grid, {
      alignContent: "between"
    }), /*#__PURE__*/React.createElement(Grid, {
      alignContent: "around"
    }), /*#__PURE__*/React.createElement(Grid, {
      alignContent: "end"
    }), /*#__PURE__*/React.createElement(Grid, {
      alignContent: "stretch"
    })));
    var tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('gap renders', function () {
    var component = renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(Grid, {
      gap: "small"
    }), /*#__PURE__*/React.createElement(Grid, {
      gap: "medium"
    }), /*#__PURE__*/React.createElement(Grid, {
      gap: "large"
    }), /*#__PURE__*/React.createElement(Grid, {
      gap: {
        row: 'small'
      }
    }), /*#__PURE__*/React.createElement(Grid, {
      gap: {
        row: 'medium'
      }
    }), /*#__PURE__*/React.createElement(Grid, {
      gap: {
        row: 'large'
      }
    }), /*#__PURE__*/React.createElement(Grid, {
      gap: {
        column: 'small'
      }
    }), /*#__PURE__*/React.createElement(Grid, {
      gap: {
        column: 'medium'
      }
    }), /*#__PURE__*/React.createElement(Grid, {
      gap: {
        column: 'large'
      }
    }), /*#__PURE__*/React.createElement(Grid, {
      gap: {
        row: 'small',
        column: 'medium'
      }
    })));
    var tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('fill renders', function () {
    var component = renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(Grid, {
      fill: true
    }), /*#__PURE__*/React.createElement(Grid, {
      fill: false
    }), /*#__PURE__*/React.createElement(Grid, {
      fill: "horizontal"
    }), /*#__PURE__*/React.createElement(Grid, {
      fill: "vertical"
    })));
    var tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('responsive', function () {
    var component = renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(Grid, {
      responsive: true
    }), /*#__PURE__*/React.createElement(Grid, {
      responsive: false
    })));
    var tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('as renders', function () {
    var component = renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(Grid, {
      as: "article"
    })));
    var tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('proxies tag', function () {
    var tagComponent = renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(Grid, {
      tag: "article"
    })));
    var asComponent = renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(Grid, {
      as: "article"
    })));
    expect(tagComponent.toJSON()).toEqual(asComponent.toJSON());
  });
  test('pad', function () {
    var component = renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(Grid, {
      pad: "small"
    }), /*#__PURE__*/React.createElement(Grid, {
      pad: "medium"
    }), /*#__PURE__*/React.createElement(Grid, {
      pad: "large"
    }), /*#__PURE__*/React.createElement(Grid, {
      pad: {
        horizontal: 'small'
      }
    }), /*#__PURE__*/React.createElement(Grid, {
      pad: {
        vertical: 'small'
      }
    }), /*#__PURE__*/React.createElement(Grid, {
      pad: {
        bottom: 'small'
      }
    }), /*#__PURE__*/React.createElement(Grid, {
      pad: {
        left: 'small'
      }
    }), /*#__PURE__*/React.createElement(Grid, {
      pad: {
        right: 'small'
      }
    }), /*#__PURE__*/React.createElement(Grid, {
      pad: {
        start: 'small'
      }
    }), /*#__PURE__*/React.createElement(Grid, {
      pad: {
        end: 'small'
      }
    }), /*#__PURE__*/React.createElement(Grid, {
      pad: {
        top: 'small'
      }
    })));
    var tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});