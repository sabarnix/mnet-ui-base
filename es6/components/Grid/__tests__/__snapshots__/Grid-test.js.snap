// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid a11yTitle renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
}

<div>
  <div
    class="c0"
  >
    <div
      aria-label="My Grid"
      class="c1"
    />
  </div>
</div>
`;

exports[`Grid align renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c2 {
  display: grid;
  box-sizing: border-box;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  display: grid;
  box-sizing: border-box;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c4 {
  display: grid;
  box-sizing: border-box;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
  <div
    className="c4"
  />
</div>
`;

exports[`Grid alignContent renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
}

.c2 {
  display: grid;
  box-sizing: border-box;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
}

.c3 {
  display: grid;
  box-sizing: border-box;
  -webkit-align-content: space-between;
  -ms-flex-line-pack: space-between;
  align-content: space-between;
}

.c4 {
  display: grid;
  box-sizing: border-box;
  -webkit-align-content: space-around;
  -ms-flex-line-pack: space-around;
  align-content: space-around;
}

.c5 {
  display: grid;
  box-sizing: border-box;
  -webkit-align-content: flex-end;
  -ms-flex-line-pack: end;
  align-content: flex-end;
}

.c6 {
  display: grid;
  box-sizing: border-box;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
  <div
    className="c4"
  />
  <div
    className="c5"
  />
  <div
    className="c6"
  />
</div>
`;

exports[`Grid areas renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  grid-template-areas: "header main footer" "header sidebar footer" ". .";
  grid-template-columns: 75% 25%;
  grid-template-rows: 32px 256px 64px;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
</div>
`;

exports[`Grid areas renders when given an array of string arrays 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  grid-template-areas: "header header" "sidebar main" "footer footer";
  grid-template-columns: 75% 25%;
  grid-template-rows: 32px 256px 64px;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
</div>
`;

exports[`Grid as renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
}

<div
  className="c0"
>
  <article
    className="c1"
  />
</div>
`;

exports[`Grid columns renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  grid-template-columns: 50% 50%;
}

.c2 {
  display: grid;
  box-sizing: border-box;
  grid-template-columns: 33.33% 66.66%;
}

.c3 {
  display: grid;
  box-sizing: border-box;
  grid-template-columns: 25% 75%;
}

.c4 {
  display: grid;
  box-sizing: border-box;
  grid-template-columns: repeat( auto-fill,minmax(128px,1fr) );
}

.c5 {
  display: grid;
  box-sizing: border-box;
  grid-template-columns: repeat( auto-fit,minmax(128px,1fr) );
}

.c6 {
  display: grid;
  box-sizing: border-box;
  grid-template-columns: repeat( auto-fill,minmax(128px,256px) );
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
  <div
    className="c4"
  />
  <div
    className="c5"
  />
  <div
    className="c6"
  />
</div>
`;

exports[`Grid fill renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  width: 100%;
  height: 100%;
}

.c2 {
  display: grid;
  box-sizing: border-box;
}

.c3 {
  display: grid;
  box-sizing: border-box;
  width: 100%;
}

.c4 {
  display: grid;
  box-sizing: border-box;
  height: 100%;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
  <div
    className="c4"
  />
</div>
`;

exports[`Grid gap renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  grid-gap: 4px 4px;
}

.c2 {
  display: grid;
  box-sizing: border-box;
  grid-gap: 8px 8px;
}

.c3 {
  display: grid;
  box-sizing: border-box;
  grid-gap: 16px 16px;
}

.c4 {
  display: grid;
  box-sizing: border-box;
  grid-row-gap: 4px;
}

.c5 {
  display: grid;
  box-sizing: border-box;
  grid-row-gap: 8px;
}

.c6 {
  display: grid;
  box-sizing: border-box;
  grid-row-gap: 16px;
}

.c7 {
  display: grid;
  box-sizing: border-box;
  grid-column-gap: 4px;
}

.c8 {
  display: grid;
  box-sizing: border-box;
  grid-column-gap: 8px;
}

.c9 {
  display: grid;
  box-sizing: border-box;
  grid-column-gap: 16px;
}

.c10 {
  display: grid;
  box-sizing: border-box;
  grid-row-gap: 4px;
  grid-column-gap: 8px;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
  <div
    className="c4"
  />
  <div
    className="c5"
  />
  <div
    className="c6"
  />
  <div
    className="c7"
  />
  <div
    className="c8"
  />
  <div
    className="c9"
  />
  <div
    className="c10"
  />
</div>
`;

exports[`Grid justify renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  justify-items: flex-start;
}

.c2 {
  display: grid;
  box-sizing: border-box;
  justify-items: center;
}

.c3 {
  display: grid;
  box-sizing: border-box;
  justify-items: flex-end;
}

.c4 {
  display: grid;
  box-sizing: border-box;
  justify-items: stretch;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
  <div
    className="c4"
  />
</div>
`;

exports[`Grid justifyContent renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c2 {
  display: grid;
  box-sizing: border-box;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  display: grid;
  box-sizing: border-box;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  display: grid;
  box-sizing: border-box;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c5 {
  display: grid;
  box-sizing: border-box;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c6 {
  display: grid;
  box-sizing: border-box;
  -webkit-box-pack: stretch;
  -webkit-justify-content: stretch;
  -ms-flex-pack: stretch;
  justify-content: stretch;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
  <div
    className="c4"
  />
  <div
    className="c5"
  />
  <div
    className="c6"
  />
</div>
`;

exports[`Grid pad 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  padding: 4px;
}

.c2 {
  display: grid;
  box-sizing: border-box;
  padding: 8px;
}

.c3 {
  display: grid;
  box-sizing: border-box;
  padding: 16px;
}

.c4 {
  display: grid;
  box-sizing: border-box;
  padding-left: 4px;
  padding-right: 4px;
}

.c5 {
  display: grid;
  box-sizing: border-box;
  padding-top: 4px;
  padding-bottom: 4px;
}

.c6 {
  display: grid;
  box-sizing: border-box;
  padding-bottom: 4px;
}

.c7 {
  display: grid;
  box-sizing: border-box;
  padding-left: 4px;
}

.c8 {
  display: grid;
  box-sizing: border-box;
  padding-right: 4px;
}

.c9 {
  display: grid;
  box-sizing: border-box;
  padding-inline-start: 4px;
}

.c10 {
  display: grid;
  box-sizing: border-box;
  padding-inline-end: 4px;
}

.c11 {
  display: grid;
  box-sizing: border-box;
  padding-top: 4px;
}

@media only screen and (max-width:512px) {
  .c1 {
    padding: 4px;
  }
}

@media only screen and (max-width:512px) {
  .c2 {
    padding: 8px;
  }
}

@media only screen and (max-width:512px) {
  .c3 {
    padding: 16px;
  }
}

@media only screen and (max-width:512px) {
  .c4 {
    padding-left: 4px;
    padding-right: 4px;
  }
}

@media only screen and (max-width:512px) {
  .c5 {
    padding-top: 4px;
    padding-bottom: 4px;
  }
}

@media only screen and (max-width:512px) {
  .c6 {
    padding-bottom: 4px;
  }
}

@media only screen and (max-width:512px) {
  .c7 {
    padding-left: 4px;
  }
}

@media only screen and (max-width:512px) {
  .c8 {
    padding-right: 4px;
  }
}

@media only screen and (max-width:512px) {
  .c9 {
    padding-inline-start: 4px;
  }
}

@media only screen and (max-width:512px) {
  .c10 {
    padding-inline-end: 4px;
  }
}

@media only screen and (max-width:512px) {
  .c11 {
    padding-top: 4px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
  <div
    className="c4"
  />
  <div
    className="c5"
  />
  <div
    className="c6"
  />
  <div
    className="c7"
  />
  <div
    className="c8"
  />
  <div
    className="c9"
  />
  <div
    className="c10"
  />
  <div
    className="c11"
  />
</div>
`;

exports[`Grid renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
</div>
`;

exports[`Grid responsive 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c1"
  />
</div>
`;

exports[`Grid rows renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: grid;
  box-sizing: border-box;
  grid-template-rows: 128px 512px 256px;
}

.c2 {
  display: grid;
  box-sizing: border-box;
  grid-auto-rows: 128px;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
</div>
`;
